### PR TITLE
[SPARK-46544][SQL] Support v2 DESCRIBE TABLE EXTENDED with table stats

### PR DESCRIPTION
**Pull request by Zouxxyy**
### What changes were proposed in this pull request?

Support v2 DESCRIBE TABLE EXTENDED with table stats

### Why are the changes needed?

Similar to  #40058, make DS v1/v2 command parity, e.g.

DESC EXTENDED table

| col_name          | data_type                 | comment    |
|-------------------|---------------------------|------------|
| ...               | ...                       | ...        |
| Statistics        | 864 bytes, 2 rows         |            |
| ...               | ...                       | ...        |

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

add test `describe extended table with stats`

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by MaxGekk** [comment]
@Zouxxyy Please, trigger GitHub actions.

**Review by MaxGekk — COMMENTED** [review]
Please, fix the failed test, seems it is related to your changes:
```
[info]   == Results ==
[info]   !== Correct Answer - 5 ==           == Spark Answer - 6 ==
[info]   !struct<>                           struct<col_name:string,data_type:string,comment:string>
[info]    [# Column Default Values,,]        [# Column Default Values,,]
[info]    [# Detailed Table Information,,]   [# Detailed Table Information,,]
[info]    [# Metadata Columns,,]             [# Metadata Columns,,]
[info]   ![id,bigint,42]                     [Statistics,0 bytes, 0 rows,null]
[info]   ![id,bigint,null]                   [id,bigint,42]
[info]   !                                   [id,bigint,null] (QueryTest.scala:267)
```

**Comment by Zouxxyy** [comment]
@MaxGekk Thanks for your reminder, fix it

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/DescribeTableSuite.scala [hunk 1]
```diff
@@ -17,6 +17,8 @@
 
 package org.apache.spark.sql.execution.command.v2
 
+import org.junit.jupiter.api.Assertions
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/DescribeTableSuite.scala:20]: Why cannot you use `assert` from scalatest?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/DescribeTableSuite.scala [hunk 2]
```diff
@@ -196,4 +199,20 @@ class DescribeTableSuite extends command.DescribeTableSuiteBase
...
+        s"""
+           |CREATE TABLE $tbl
+           |(key INT, col STRING)
+           |$defaultUsing""".stripMargin)
+
+      sql(s"INSERT INTO $tbl values (1, 'aaa'), (2, 'bbb'), (3, 'ccc'), (null, 'ddd')")
+      val descriptionDf = sql(s"DESCRIBE TABLE EXTENDED $tbl")
+      val stats = descriptionDf.filter("col_name == 'Statistics'").select("data_type")
+        .collectAsList().get(0).toString
+      Assertions.assertTrue("""\[\d+\s+bytes,\s+4\s+rows]""".r.matches(stats))
```
Comment (MaxGekk) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/command/v2/DescribeTableSuite.scala:215]:
```suggestion
      assert("""\[\d+\s+bytes,\s+4\s+rows]""".r.matches(stats))
```

**Review by MaxGekk — CHANGES_REQUESTED** [review]

**Comment by Zouxxyy** [comment]
@MaxGekk Update, thanks

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @Zouxxyy.

**Pull request closed without merging**